### ISSUE TITLE
Launch subprocesses to speed things up

### ISSUE THREAD

**Issue by tdgroot** [elgentos/masquerade#24]
Instead of running everything in the same php thread, it'd be nice if the application runs a subprocess for each table or something like that.

**Comment by peterjaap** [comment]
What would you propose we use for threading? pthreads is only safe in PHP 7.2+ which will restrict usage to only 7.2, which isn't very widely adopted yet.

Another but hacky way would be to let masquerade call itself through Symfony Process + one of the Parallel Process packages.

**Comment by peterjaap** [comment]
I've done some preliminary work in the [24-subprocesses](https://github.com/elgentos/masquerade/tree/24-subprocesses) branch. See commit https://github.com/elgentos/masquerade/commit/09ce2ac37afc297360b33710e9d4581636daf.

It works, but it immediately outputs the progressbar for every running process, making a mess;

```
 _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1
                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1

Updating admin_user
     0/26750 [>---------------------------]   0%                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1

Updating sales_creditmemo

Updating email_contact

Updating newsletter_subscriber
  0/14 [>---------------------------]   0%   0/280 [>---------------------------]   0%     0/11616 [>---------------------------]   0%     0/11670 [>---------------------------]   0%                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1
                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1
                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1

Updating sales_invoice

Updating review_detail
    0 [>---------------------------]
Done anonymizing
     0/32941 [>---------------------------]   0%                              
._ _  _. _ _.    _ .__. _| _  
| | |(_|_>(_||_|(/_|(_|(_|(/_ 
            |
                   by elgentos
                        v0.1.1

Updating sales_order
     0/38379 [>---------------------------]   0%
Updating quote
     0/78297 [>---------------------------]   0%

Done anonymizing
 14/14 [============================] 100%^
```

**Comment by tdgroot** [comment]
Cool stuff! It might be better to work with return values in the subprocesses, read them in the master process and create progress bars based on that. 

Not sure if that's possible, this is new stuff for me haha.

**Comment by tdgroot** [comment]
FWIW, I wrote a little bash script to make it possible for now:
```bash
DATABASE="your_db_name"
MASQUERADE_PLATFORM="magento2"
MASQUERADE_GROUPS=($(bin/masquerade groups --platform "${MASQUERADE_PLATFORM}" | grep '|' | grep -v 'Group' | awk -F'|' '{print $3}' | uniq))

for group in "${MASQUERADE_GROUPS[@]}"
do
    echo "Starting process for group $group"
    screen -d -m -S "anonymize_${group}" bin/masquerade run --platform "${MASQUERADE_PLATFORM}" --database "${DATABASE}" --group "${group}"
done
```

**Comment by peterjaap** [comment]
I created a subprocess per group/table combi;

![image](https://user-images.githubusercontent.com/431360/67004841-7bfd0c80-f0e1-11e9-8da6-000772f5527f.png)

**Comment by peterjaap** [comment]
@tdgroot could you test it?

You can clone the https://github.com/elgentos/masquerade/tree/24-subprocesses branch and run `bin/masquerade` to test it

**Comment by peterjaap** [comment]
This is a nicer package with some more options; https://github.com/graze/parallel-process

**Comment by peterjaap** [comment]
The main problem this issue tried to solve was speed, and that was fixed in version 0.3.0. So closing this issue.
